.github/workflows/ci.yml

```diff
@@ -1,115 +1,57 @@
 name: CI
 
 on:
-    push:
-        branches:
-          - develop
     pull_request:
         branches:
           - develop
 
+env:
+  UV_SYSTEM_PYTHON: true
+
 jobs:
-  test:
+  ci:
     runs-on: ubuntu-latest
-
-    strategy:
-      fail-fast: false
-      matrix:
-        python-version: ["3.12"]
```

Comment:
We need to keep the matrix. Actually we should be adding 3.13 to it, but not as part of this CR